Packs/MicrosoftExchangeOnline/Integrations/EWSO365/EWSO365.py

```diff
@@ -2158,17 +2195,28 @@ def parse_incident_from_item(item):     # pragma: no cover
 
                             v = ' '.join(map(str.strip, v.split('\r\n')))
                             attached_email_headers.append((h.lower(), v))
+                        demisto.debug(f'{attached_email_headers=}')
                         for header in attachment.item.headers:
+                            demisto.debug(f'{header.name.lower()}, {header.value}')
                             if (
                                     (header.name.lower(), header.value)
                                     not in attached_email_headers
                                     and header.name.lower() != "content-type"
                             ):
-                                try:
-                                    attached_email.add_header(header.name, header.value)
-                                except ValueError as err:
-                                    if "There may be at most" not in str(err):
-                                        raise err
+                                if header.name.lower() == "message-id":
+                                    # add the message-id to the headers
+                                    demisto.debug('adding message-id to the headers')
+                                    try:
+                                        attached_email.add_header(header.name, header.value)
+                                    except ValueError as err:
+                                        if "There may be at most" not in str(err):
+                                            raise err
+                                else:
+                                    try:
+                                        attached_email.add_header(header.name, header.value)
+                                    except ValueError as err:
+                                        if "There may be at most" not in str(err):
+                                            raise err
```

Comment:
What's the difference between the `if`/`else` branches?

Reply:
I made a mistake here by not removing this redundant code.
The initial commit in this PR had two functions:

1. `handle_attached_email_with_incorrect_id` which recived `attached_email` as an argument and handled the wrong "Message-ID" header value before the for loop over `attached_email.items()` [line 2187].
2. And another function that was called `handle_incorrect_id` which performed the same logic as the first function but later in the code here:
`attached_email.add_header(header.name, handle_incorrect_id(header.value))` [previously line 2210].
This is redundant due to `handle_attached_email_with_incorrect_id` which handles this earlier in the code, meaning we can return this code segment to the original logic.